exportModule.js

```diff
@@ -1,30 +1,29 @@
+const fs = require('fs');
 const data = require('./response.json');
 const loadData = require('./GetDataType.js');
 const states = require('./states.js');
-const stateLookup = require('./stateMap.js');
-function exportModule(data) {
-    const dataTypes = loadData();
-    const moduleJSON = {
-        "name": data.library.identifier.id,
-        "remarks": [],
-        "states": {
-            "Initial": new states.InitialState('Initial').toJSON(),
-            "Terminal": new states.TerminalState().toJSON()
-        }
-    };
-    dataTypes.forEach(function (temp, i) {
-        try {
-            const map = stateLookup[temp];
-            const stateName = `${temp.substring(21)}_${i}`;
-            const state = new map(stateName, [], {})
-            moduleJSON.states[stateName] = state.toJSON();
-        }
-        catch{
-            console.log('no corresponding state for that data type');
-            return;
-        }
-    })
-    return moduleJSON
+const factory = require('./factory.js');
+
+function exportModule() {
+  const dataTypes = loadData();
+  const moduleJSON = {
+    name: data.library.identifier.id,
+    remarks: [],
+    states: {
+      Initial: new states.InitialState('Initial').toJSON(),
+      Terminal: new states.TerminalState().toJSON(),
+    },
+  };
+  dataTypes.forEach(function (temp, i) {
+    try {
+      const stateName = `${temp.substring(21)}_${i}`;
+      moduleJSON.states[stateName] = factory(temp).toJSON();
+    } catch {
+      return;
+    }
```

Comment:
I think we might need to restructure this. I don't think the Try/catch is working like we think it should. I'd recommend doing something like this:


```suggestion
  const stateName = `${temp.substring(21)}_${i}`;
  const StateClass = factory(temp);
  if (StateClass !== null) {
    moduleJSON.states[stateName] = factory(temp).toJSON();
  }
```

This is under the assumption that we update the return value in the `default` case of the switch statement to be `null`